UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -18,21 +19,32 @@ public abstract class Brain : ScriptableObject
     {
         [SerializeField] public BrainParameters brainParameters;
 
-        protected Dictionary<Agent, AgentInfo> m_AgentInfos =
-            new Dictionary<Agent, AgentInfo>(1024);
+        protected List<Agent> m_AgentInfos = new List<Agent>(1024);
 
-        [System.NonSerialized]
+        protected ICommunicator m_Communicator;
+
+        [NonSerialized]
         private bool m_IsInitialized;
 
+        /// <summary>
+        /// Sets the Batcher of the Brain. The brain will call the communicator at every step and give
+        /// it the agent's data using PutObservations at each DecideAction call.
+        /// </summary>
+        /// <param name="communicator"> The Batcher the brain will use for the current session</param>
+        public void SetCommunicator(ICommunicator communicator)
+        {
+            m_Communicator = communicator;
+            LazyInitialize();
+        }
+
         /// <summary>
         /// Adds the data of an agent to the current batch so it will be processed in DecideAction.
         /// </summary>
         /// <param name="agent"></param>
-        /// <param name="info"></param>
-        public void SendState(Agent agent, AgentInfo info)
+        public void SendState(Agent agent)
```

Comment:
This also feels like a bad name since it's not sending anything.

Reply:
change the name to `Update`

Reply:
update is a bad name.

Reply:
Named it `SubscribeAgentForDecision`